rasa/core/nlg/interpolator.py

```diff
@@ -53,3 +53,23 @@ def interpolate(
                 template[k] = interpolate_text(v, values)
         return template
     return template
+
+
+def interpolate_button_text(template: Text, values: Dict[Text, Text]) -> Text:
+    for slot in values:
+        if values[slot]:
+            template = template.replace("{" + str(slot) + "}", values[slot])
+    return template
+
+
+def interpolate_buttons(template: List[Dict], values: Dict[Text, Text]) -> List[Dict]:
+    if isinstance(template, list):
+        for idx, a_button_dict in enumerate(template):
+            if isinstance(a_button_dict, dict):
+                template[idx]["payload"] = interpolate_button_text(
+                    a_button_dict["payload"], values
+                )
+                template[idx]["title"] = interpolate_button_text(
+                    a_button_dict["title"], values
+                )
+    return template
```

Comment:
that's not needed, right?

Reply:
These are needed, the original code won't work otherwise.

Reply:
@8key Got the problem now. I think we should handle this more general and not just for buttons. The problem is that we are currently interpolating `dict` and `string` items, but no `list`s. 

This should fix it for all sorts of keys:

```python
def interpolate(
    template: Union[List[Any], Dict[Text, Any], Text], values: Dict[Text, Text]
) -> Union[List[Any], Dict[Text, Any], Text]:
    if isinstance(template, str):
        return interpolate_text(template, values)
    elif isinstance(template, dict):
        for k, v in template.items():
            if isinstance(v, dict):
                interpolate(v, values)
            elif isinstance(v, list):
                template[k] = [interpolate(i, values) for i in v]
            else:
                template[k] = interpolate_text(v, values)
        return template
    # This is the added the case which we missed before
    elif isinstance(template, list):   
        return [interpolate(i, values) for i in template]

    return template
```

Reply:
Okay, make sense, will update thanks

Reply:
Thanks for fixing this 🚀 🙏

Reply:
Added this line, would have a missing key error otherwise.
this is because
button template has value e.g., 
`/choose{"appt_time": "{candidate_appt_time_1}"}`
need it to be
`/choose{{"appt_time": "{candidate_appt_time_1}"}}`
that the outer curly braces have to be escaped while apply `.format` method
https://github.com/8key/rasa/blob/7773e40427c251a0b762f61314fa8cc6cf169675/rasa/core/nlg/interpolator.py#L25

Reply:
let me know if anything